CONTRIBUTING.md

```diff
@@ -49,15 +49,88 @@ If your pull request contains any of the below, it's less likely to be merged:
+    - DependencyAnalzyers are covered in more depth in [extensibility dependency analyzer docs](./docs/extensibility.md#dependency-analyzers).
+  - [`PackageUpdaterPreTFMStep`](src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageUpdaterPreTFMStep.cs) is just like `PackageUpdaterStep` except that it's run prior to updating the project's TFM, so it will remove transitive dependencies but likely won't change any package versions. This step is meant as a way to clean up package references for users who want to modernize their project files without completely upgrading to .NET Core or .NET 5/6.
+  - [`TemplateInserterStep`](src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Templates/TemplateInserterStep.cs) adds template files appropriate for the project type to the project being upgraded if they don't exist yet.
+  - [`SourceUpdaterStep`](src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source/SourceUpdaterStep.cs) uses Roslyn analyzers and code fix providers to identify code patterns in C# and VB files that will need updated as part of the upgrade process.
+  - [`ConfigUpdaterStep`](src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration/ConfigUpdaterStep.cs) uses registered `IUpdater<ConfigFile>` instances to apply project-level upgrades based on the contents of web.config and app.config files.
+    - [`IUpdater<ConfigFile>`](src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpdater.cs) is covered in more depth in [extensibility docs](docs/extensibility.md#updaters).
+  - [`RazorUpdaterStep`](src/extensions/web/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/RazorUpdaterStep.cs) updates Razor views (cshtml files) in the project using registered `IUpdater<RazorCodeDocument>` implementations.
+    - [`IUpdater<RazorCodeDocument>`](src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpdater.cs) is covered in more depth in [extensibility docs](docs/extensibility.md#updaters).
+  - [`NextProjectStep`](src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/NextProjectStep.cs) runs after upgrading a specific projects is complete and clears the current project from the upgrade context. This will cause most project-specific steps to reset their state the next time the UpgraderManager checks their applicability, causing them to run again on the next project the user selects.
+  - [`FinalizeSolutionStep`](src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/FinalizeSolutionStep.cs) is intended to run as the last step of the upgrade process and performs cleanup activities such as deleting the .upgrade-assistant state file.
+
+### Other services
+
+- [`IUpgradeReadyCheck`](src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpgradeReadyCheck.cs) implementations are used by the `CurrentProjectSelectionStep` to validate that projects are able to be meaningfully upgraded by Upgrade Assistant and can be successfully loaded.
```

Comment:
It would be nice to have a section of ready checks with a list of things that we've checked for

Reply:
Good idea. I'll add that.